### PR TITLE
Monorepo: updates package creation docs

### LINKED ISSUE #44277: (2P) Update monorepo docs to use yarn instead of npm in "developing" section and mention CSS & SCSS as sideEffects
[Currently the Monorepo docs reference `npm link` where they should say `yarn link`](https://github.com/Automattic/wp-calypso/blob/9be576d57f6c105b1845973cd7f2dba2f0af6efd/docs/monorepo.md#developing-packages). Update the docs accordingly.

Also add in a note about needing to use `copy-assets` if importing Sass into JS (see p1595244326057000-slack-calypso-framework).

```js
"prepare": "check-npm-client && transpile && copy-assets"
```

Also include note about ensuring dependencies are correctly listed in the `package.json`. This can be validated by:

```
yarn workspace @automattic/{{YOUR_PACKAGE_HERE}} run prepare && npx @yarnpkg/doctor packages/{{YOUR_PACKAGE_HERE}}
```

Also include information about [specifying CSS and SCSS as `sideEffects` to ensure they aren't removed via tree shaking.](https://github.com/material-components/material-components-web/issues/4702). See also [here](https://github.com/Automattic/wp-calypso/blob/e8e49f9b1b5cab4679441c005f493ad4444765c4/packages/components/package.json#L10-L13).

### PR DESCRIPTION
**Pull request by cpapazoglou**
#### Changes proposed in this Pull Request
Update the monorepo documentation:
- elaborate on `sideEffects`
- add `dependencies` guidelines
- add `copy-assets` to prepare script (if needed)
- change `npm link` to `yarn link`

Fixes #44277

### CODE REVIEW

File: docs/monorepo.md [hunk 1]
```diff
@@ -38,8 +38,30 @@ test/
...
 Your `package.json` can have any of the [normal properties](https://docs.npmjs.com/files/package.json) but at a minimum should contain `main`, `module`, and `sideEffects`.
 
+### devDependencies
+
 It used to be that `devDependencies` needed to be added to the root `package.json` but since we moved to `yarn` workspaces we're able to add them as regular `devDependencies` within the package that uses them.
 
+### dependencies
+
+Running the following in your `wp-calypso` root <br/>
+`yarn workspace @automattic/{{your-package}} run prepare && npx @yarnpkg/doctor packages/{{your-package}}` <br/>
```
Comment (retrofox) [inline docs/monorepo.md:48]: is `<br />` a valid/proper markdown to use here? I think it's better just to apply line breaks.

Comment (retrofox) [reply docs/monorepo.md:48]: I see only two `<br>`s in the whole doc, just fyi.

Comment (cpapazoglou) [reply docs/monorepo.md]: Great observation! Actually I am using a markdown previewer and it looked good, but I searched it a little more and....
TIL: to add a line break in .md add two or more spaces and then hit return (one space & return won't work).
https://github.com/Automattic/wp-calypso/pull/44481/commits/f22d36be22c8067fa74b35c0878056044a48b6fa

Thanks @retrofox !

**Comment by getdave** [comment]
@cpapazoglou Love this. I was about to start following up on this Issue and then I discovered you'd already done it! Thanks so much for following up!

**Pull request merged**